src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
+        resizeArray();
+        array[size] = value;
+        size++;
 
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index value."
+                    + " No such index or index is less than 0");
+        }
+        resizeArray();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list != null) {
+            for (int i = 0; i < list.size(); i++) {
```

Comment:
for-each cycle will look more tidy

Reply:
Do we need to make some extra import? Because if we use for-each loop with "import core.basesyntax.ArrayList;", it create some mistake. And "import java.util.ArrayList;" we can`t make.